web/js/modules/product-picker/format-config.js

```diff
@@ -122,7 +123,10 @@ export default function buildLayerFacetProps(config, selectedDate) {
   return lodashMap(layers, (layer) => {
     setCoverageFacetProp(layer, selectedDate);
     setLayerPeriodFacetProps(layer);
-    if (layer.daynight && layer.daynight.length) {
+    if (layer.daynight && layer.daynight.length && layer.daynight.map) {
+      if (typeof layer.daynight === 'string') {
+        layer.daynight = [layer.daynight];
+      }
```

Comment:
Minor possible suggestion (slightly less verbose):
```suggestion
      if (layer.daynight && !Array.isArray(layer.daynight) {
        layer.daynight = [layer.daynight];
      }
```

Reply:
I guess my question about this field is... should we allow non-array values?  I think it's fine since we can convert it to an array like this but, I guess I was intending it to always be an array in the configs.  

If you enter a non-array value for the `tracks` property in the configs, things break as well
![Screen Shot 2020-06-12 at 4 06 32 PM](https://user-images.githubusercontent.com/1480000/84542120-c792a500-acc6-11ea-8a80-c3a0298e9c20.png)

Reply:
I guess what I'm getting at is, maybe what would be better is more accurate type checking on the configs themselves at build time?

Reply:
Hmm, I didn't mean to include `&& layer.daynight.map`. That overrides the logic I wrote below it, but,  I think you still need the two sets of 'if' statements.

Reply:
I totally agree (referenced `Typescript` in description area. I would love to go that route.). I just added this to prevent something potentially breaking on master in the comming release. It definitely was breaking on a non-production config I was just using. I'm not sure if there are layers on master that have this field as a string, but I imagine there might be.

Reply:
Should we create a ticket to add some sort of type checking, whether that be in the form of typescript or something else at build-time?

Reply:
Since this `daynight` property will likely move into the yet-to-be-named visualization metadata service, I wouldn't worry about creating a ticket for now.  I think just doing some safety checks and casting it to an array on the client side is fine for now 👍